samtranslator/intrinsics/resource_refs.py

```diff
@@ -13,14 +13,15 @@ def __init__(self) -> None:
         # { "LogicalId": {"Property": "Value"} }
         self._refs: Dict[str, Dict[str, Any]] = {}
 
-    def add(self, logical_id, property_name, value):  # type: ignore[no-untyped-def]
+    def add(self, logical_id: str, property_name: str, value: str) -> None:
         """
         Add the information that resource with given `logical_id` supports the given `property`, and that a reference
         to `logical_id.property` resolves to given `value.
 
         Example:
 
             "MyApi.Deployment" -> "MyApiDeployment1234567890"
+            "SuperCoolAPI.DataSources:MyDataSource" -> "SuperCoolAPIMyDataSource"
```

Comment:
How necessary is this?

The `.` notation is useful for things that cannot be easily determined, such as the example above where the deployment includes a hash. For everything else, we document the generated logical IDs, so I'm not sure whether it's worth adding a new concept to refer to resources.